sceptre/connection_manager.py

```diff
@@ -122,27 +132,102 @@ def __repr__(self):
+        return self._get_session(profile, region, iam_role)
+
+    def create_session_environment_variables(
+        self,
+        profile: Optional[str] = STACK_DEFAULT,
+        region: Optional[str] = STACK_DEFAULT,
+        iam_role: Optional[str] = STACK_DEFAULT,
+        include_system_envs: bool = False,
+    ) -> Dict[str, str]:
+        """Creates the standard AWS environment variables that would need to be passed to a
+        subprocess in a hook, resolver, or template handler and allow that subprocess to work with
+        the currently configured session.
+
+        :param profile: The name of the AWS Profile as configured in the local environment. Passing
+            None will result in no profile being specified. Defaults to the ConnectionManager's
+            configured profile (if there is one).
+        :param region: The AWS Region the session should be configured with. Defaults to the
+            ConnectionManager's configured region.
+        :param iam_role: The IAM role ARN that is assumed using STS to create the session. Passing
+            None will result in no IAM role being assumed. Defaults to the ConnectionManager's
+            configured iam_role (if there is one).
+        :param include_system_envs: If True, will return a dict with all the system environment
+            variables included. This is useful for creating a complete dict of environment variables
+            to pass to a subprocess. By default, this method will ONLY return the relevant AWS
+            environment variables.
+
+        :returns: A dict of environment variables with the appropriate credentials available for use.
+        """
+        session = self.get_session(profile, region, iam_role)
+        # Set aws environment variables specific to whatever AWS configuration has been set on the
+        # stack's connection manager.
+        credentials: Credentials = session.get_credentials()
+        envs = dict(**self._get_envs()) if include_system_envs else {}
+
+        if include_system_envs:
+            # We don't want a profile specified, since that could interfere with the credentials we're
+            # about to set. Even if we're using a profile, the credentials will already reflect that
+            # profile's configurations.
+            envs.pop("AWS_PROFILE", None)
+
+        envs.update(
+            AWS_ACCESS_KEY_ID=credentials.access_key,
+            AWS_SECRET_ACCESS_KEY=credentials.secret_key,
+            # Most AWS SDKs use AWS_DEFAULT_REGION for the region; some use AWS_REGION
+            AWS_DEFAULT_REGION=session.region_name,
+            AWS_REGION=session.region_name,
+        )
+
+        if credentials.token:
+            envs["AWS_SESSION_TOKEN"] = credentials.token
+        # There might not be a session token, so if there isn't one, make sure it doesn't exist in
+        # the envs being passed to the subprocess
+        elif include_system_envs:
+            envs.pop("AWS_SESSION_TOKEN", None)
```

Comment:
Don't we want the subprocess to execute while assuming a role?  if so i'm not sure why we want to remove the session token?

Reply:
This is to account for what might be set in the local user's environment variables.

For example, let's say the local user already has a session token set as a part of creating a session using cli tooling. (We do this at my work). That environment variable might be already set on the process. 

However, if we've created a session, we want to use THAT session's credentials (key id, secret key, and token if there is one). The new session's credentials will likely be different from what the user's logged in session will be.
* We already always overwrite the key id and secret from whatever the OS environment variables are. If they're the same values, great. If not, we overwrite them and it'll just work.
* If the credentials produced from the new session HAVE a token, we'll make sure that's the token we use. If it's the same as in the OS, great. If not, we'll overwrite it.
* HOWEVER, if the OS has a session token environment variable set, but the new session's credentials don't have one, then we don't want that session token environment variable set. This prevents passing the session token along for credentials that should not have a session token set.

Anyway, we're making the decision based upon whether or not the session _that we want to use_ has a token, not whether the environment variables of the current process has a token.

Reply:
> This is to account for what might be set in the local user's environment variables.  For example, let's say the local user already has a session token set as a part of creating a session using cli tooling. (We do this at my work). That environment variable might be already set on the process.

ahh ok, i didn't think of that situation. Could you please describe this in the `create_session_environment_variables` doc string or the faq?

Reply:
Yeah. I'll do that in the docstring, since that's a pretty low-level detail. In the end, this "just works like it should". The goal here was to abstract away all those pesky nit-picky situation-specific details so the user can just get the dict of envs they need.